### DISCUSSION TITLE
Big gap in the GraphQL implementation: Working with M2M in the update_item mutation

### DISCUSSION THREAD

**Discussion by verystack** [directus/directus#14699] [Feature Requests]
```graphql
mutation($id: ID) {
         update_SomeCollection_item(id: "1", data: {
                  someM2MField: [{ related_SomeCollection_id: $id }]
         }) {
                   id
          }
}
```

The above is impossible. It shouldn't be. There's no ability to create/set M2M IDs for primary item/foreign item on the fly in the update_X_item mutation. You are limited to updating/removing Junctions by the actual Junction ID. This strikes me as a pretty poor design choice.

It means: 
1. You have to run multiple mutations in common scenarios - (in an App we're working on, all data (incl. M2M + M2O) is presented as forms - which then all get saved at the one time. Having the user update M2M thru a separate/immediate event (like happens on the Directus app) does not gel with this common way of working).
2. It doesn't align with how relations work elsewhere - why can M2O fields be updated in a singular mutation with the rest of the fields, but M2M cannot? (Of course I know the reason - but the way this works breaks consistency).
3. Interacting directly with 'hidden' union types breaks the 'hands-off' nature of working with Directus for developers.
4. Potential orphaned junctions over time? I'm not 100% sure on this so I don't know at what point in the lifecycle junctions are removed - but can see it being a potential problem.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
